server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java

```diff
@@ -45,12 +57,40 @@ public final class TaskExecutionTimeTrackingEsThreadPoolExecutor extends EsThrea
         ThreadFactory threadFactory,
         RejectedExecutionHandler handler,
         ThreadContext contextHolder,
-        TaskTrackingConfig trackingConfig
+        TaskTrackingConfig trackingConfig,
+        MeterRegistry meterRegistry
     ) {
-        super(name, corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, threadFactory, handler, contextHolder);
+        super(
+            name.toCompositeString(),
+            corePoolSize,
+            maximumPoolSize,
+            keepAliveTime,
+            unit,
+            workQueue,
+            threadFactory,
+            handler,
+            contextHolder
+        );
         this.runnableWrapper = runnableWrapper;
         this.executionEWMA = new ExponentiallyWeightedMovingAverage(trackingConfig.getEwmaAlpha(), 0);
         this.trackOngoingTasks = trackingConfig.trackOngoingTasks();
+        this.queueLatencyGauge = meterRegistry.registerLongsGauge(
+            ThreadPool.THREAD_POOL_METRIC_PREFIX + name.threadPoolName() + THREAD_POOL_METRIC_NAME_QUEUE_TIME,
+            "Time tasks spent in the queue for the " + name.threadPoolName() + " thread pool",
+            "milliseconds",
+            () -> {
+                List<LongWithAttributes> metricValues = Arrays.stream(LATENCY_PERCENTILES_TO_REPORT)
+                    .mapToObj(
+                        percentile -> new LongWithAttributes(
+                            handlingTimeTracker.getPercentile(percentile / 100f),
+                            Map.of("percentile", String.valueOf(percentile))
+                        )
+                    )
+                    .toList();
+                handlingTimeTracker.clear();
+                return metricValues;
+            }
+        );
```

Comment:
I wonder if rather than publishing a time-series-per-percentile (using `percentile` attribute) we should publish a metric-per-percentile.
The metric makes no sense if you don't filter by a percentile label.

Reply:
Is it easier to plot different percentiles on the same graph with labels (and group by) compared to two different time series?

Reply:
I don't _think_ that makes a difference, but I'm not sure.

Reply:
Yes, just having a look at Kibana just now, it would be much easier to plot as a single metric grouped-by the percentiles. As separate metrics we'd need to add them as distinct time-series.